### PR TITLE
Clarified error messages around `azure_workspace_resource_id`

### LINKED ISSUE #1049: [ISSUE] Provider issue : Error: cannot read cluster: cannot configure azure-client-secret auth:
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Configuration
```hcl
# Copy-paste your Terraform configuration here
provider "databricks" {
  azure_workspace_resource_id = module.azure_databricks.databricks_id  #(This is id of azurerm_databricks_workspace.this.id)
}


resource "azurerm_databricks_workspace" "databricks" {
  depends_on = [
    azurerm_subnet_network_security_group_association.adb_public_subnet_association,
    azurerm_subnet_network_security_group_association.adb_private_subnet_association
  ]
  resource_group_name         = var.databricks_rg_name
  name                        = local.databricks_name
  location                    = var.databricks_location
  sku                         = "premium"
  managed_resource_group_name = "${var.databricks_rg_name}-adbs"
  tags                        = module.metadata_validation.resource_tags


  custom_parameters {
    private_subnet_name                                  = azurerm_subnet.private.name
    public_subnet_name                                   = azurerm_subnet.public.name
    virtual_network_id                                   = azurerm_virtual_network.vnet.id
    public_subnet_network_security_group_association_id  = azurerm_network_security_group.adb_nsg.id
    private_subnet_network_security_group_association_id = azurerm_network_security_group.adb_nsg.id
  }
}
```

- Installed hashicorp/azurerm v2.92.0 (self-signed, key ID 34365D9472D7468F)
- Installing hashicorp/azuread v2.15.0...

### Expected Behavior
What should have happened?

### Actual Behavior
What actually happened?

This was working before, we just upgraded the azure provider version and this started facing this issue, we already have resources configured in databrick using same configuration, we just went ahead and added some change (added code to create databrick token)

Error: cannot read cluster: cannot configure azure-client-secret auth: cannot get workspace: somehow resource id is not set. Environment variables used: ARM_CLIENT_SECRET, ARM_CLIENT_ID, ARM_TENANT_ID. Please check https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs#authentication for details

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Terraform and provider versions

Please paste the output of `terraform version`. If version of `databricks` provider is not the latest (https://github.com/databrickslabs/terraform-provider-databricks/releases), please make sure to use the latest one.

### Debug Output
Please add turn on logging, e.g. `TF_LOG=DEBUG terraform apply` and run command again, paste it to gist & provide the link to gist. If you're still willing to paste in log output, make sure you provide only relevant log lines with requests.

It would make it more readable, if you pipe the log through `| grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'`, e.g.:

```
TF_LOG=DEBUG terraform plan 2>&1 | grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'
```

If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Important Factoids
Are there anything atypical about your accounts that we should know?

### PR DESCRIPTION
**Pull request by nfx**
Fixed #1049

### CODE REVIEW

File: common/azure_auth.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func (aa *DatabricksClient) ensureWorkspaceURL(ctx context.Context,
 	}
 	resourceID := aa.AzureResourceID
 	if resourceID == "" {
-		return fmt.Errorf("somehow resource id is not set")
+		return fmt.Errorf("please set `azure_workspace_resource_id` provider argument")
```
Comment (alexott) [inline common/azure_auth.go:188]: Should we clarify that similar to host clarification?

**Review by alexott — APPROVED** [review]
Lgtm, just one comment

File: common/azure_auth.go (hunk 1 above)
Comment (nfx) [reply common/azure_auth.go:188]: This is when host is set and no resource id

Comment (alexott) [reply common/azure_auth.go:188]: I meant more detailed explanation, like this: https://github.com/databrickslabs/terraform-provider-databricks/blob/master/common/client.go#L278

**Pull request merged**